pkg/virt-api/webhooks/defaults.go

```diff
@@ -236,3 +237,18 @@ func setDefaultArchitecture(clusterConfig *virtconfig.ClusterConfig, spec *v1.Vi
 		spec.Architecture = clusterConfig.GetDefaultArchitecture()
 	}
 }
+
+func setDefaultGuestMemory(clusterConfig *virtconfig.ClusterConfig, spec *v1.VirtualMachineInstanceSpec) {
+	if spec.Domain.Resources.Requests.Memory().Value() == 0 &&
+		(spec.Domain.Memory == nil || spec.Domain.Memory.Guest == nil && spec.Domain.Memory.Hugepages == nil) {
+		defaultGuestMemory := clusterConfig.GetConfig().DefaultGuestMemory
+		if defaultGuestMemory == nil {
+			parsedMemory := resource.MustParse("512Mi")
```

Comment:
nit - We should move this into the actual KV CR definition with the `+default=`  marker in the future.

Reply:
For context here's some `kube-openapi` tests showing how the `+default=` marker can be used:

https://github.com/kubernetes/kube-openapi/blob/a781456278336f662dbccf613a9a725480448bce/test/integration/testdata/defaults/default.go